src/jit/codegenxarch.cpp

```diff
@@ -1948,32 +1919,34 @@ CodeGen::genCodeForTreeNode(GenTreePtr treeNode)
         break;
 
     case GT_STORE_LCL_FLD:
-        {
-            if (!genStoreRegisterReturnInLclVar(treeNode))
-            {
-                noway_assert(targetType != TYP_STRUCT);
-                noway_assert(!treeNode->InReg());
-                assert(!varTypeIsFloating(targetType) || (targetType == treeNode->gtGetOp1()->TypeGet()));
+        {            
+            noway_assert(targetType != TYP_STRUCT);
+            noway_assert(!treeNode->InReg());
+            assert(!varTypeIsFloating(targetType) || (targetType == treeNode->gtGetOp1()->TypeGet()));
 
 #ifdef FEATURE_SIMD
-                // storing of TYP_SIMD12 (i.e. Vector3) field
-                if (treeNode->TypeGet() == TYP_SIMD12)
-                {
-                    genStoreLclFldTypeSIMD12(treeNode);
-                    break;
-                }
-#endif
-
-                GenTreePtr op1 = treeNode->gtOp.gtOp1;
-                genConsumeRegs(op1);
-                emit->emitInsBinary(ins_Store(targetType), emitTypeSize(treeNode), treeNode, op1);
+            // storing of TYP_SIMD12 (i.e. Vector3) field
+            if (treeNode->TypeGet() == TYP_SIMD12)
+            {
+                genStoreLclFldTypeSIMD12(treeNode);
+                break;
             }
+#endif                
+            GenTreePtr op1 = treeNode->gtGetOp1();
+            genConsumeRegs(op1);
+            emit->emitInsBinary(ins_Store(targetType), emitTypeSize(treeNode), treeNode, op1);
         }
         break;
 
     case GT_STORE_LCL_VAR:
-        {
-            if (!genStoreRegisterReturnInLclVar(treeNode))
+        {  
+            // var = call, where call returns a multi-reg return value
+            // case is handled separately.
+            if (treeNode->IsMultiRegStoreToLocal())
```

Comment:
Since the only place this can happen is var = call, and we don't anticipate other cases for xarch, I would suggest a more meaningful name such as IsMultiRegCallStoreToLocal().